### PR TITLE
Contact page

### PR DESCRIPTION
**Pull request by SenJames**
The contact page has just being completed and ready for review.

The route is /team-086-group-a-frontend/contact

Images below:
image

image

### CODE REVIEW

**Review by chibuike07 — CHANGES_REQUESTED** [review]
the styles try and look into it, 

1. In contact.jsx, your state value should be added in the nonregistered context in the context folder
2. In the contact form, the placeholder does not appear on page load rather when the value has been added and deleted
3.  seems you prefer using APIs, why not try and do it from scratch. 
4.  In the buttons you used, A button component has been already set for everybody.
5.  The google map shows on load and immediately throws an error. check your console,
6. check the contact.jsx line 149 where you used disable it throwing an error to the console, 
7. CSS style should the defined in the style folder not any other place
8. In the unprotected routes your component paths seem not to be correct
9. again seems that the bootstrap css you added is affecting the other components css please try and check it out

**Review by HENRYKC24 — COMMENTED** [review]
@SenJames I really could not see the page when i routed to team-086-group-a-frontend/contact
Screenshot from 2020-09-18 06-00-49

**Comment by SenJames** [comment]
> the styles try and look into it,
> 
> 1. In contact.jsx, your state value should be added in the nonregistered context in the context folder
> 2. In the contact form, the placeholder does not appear on page load rather when the value has been added and deleted
> 3. seems you prefer using APIs, why not try and do it from scratch.
> 4. In the buttons you used, A button component has been already set for everybody.
> 5. The google map shows on load and immediately throws an error. check your console,
> 6. check the contact.jsx line 149 where you used disable it throwing an error to the console,
> 7. CSS style should the defined in the style folder not any other place
> 8. In the unprotected routes your component paths seem not to be correct
> 9. again seems that the bootstrap css you added is affecting the other components css please try and check it out

Hello TTL. 

Thanks for the review....

I have made the required changes except the suggestion to use API and they seem to be working now. 

Perhaps the API usage is one of choice of style. Anyway I am willing to learn. 


Kindly help review.

**Comment by SenJames** [comment]
> @SenJames I really could not see the page when i routed to team-086-group-a-frontend/contact
> Screenshot from 2020-09-18 06-00-49

Hi KC, 

You need to switch to the contact page branch to be able see the changes

Thanks.

File: src/Pages/ContactPage/contact.jsx [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+import Button from "../../Common/Button.component/Button";
+import CustomInput from "../../Common/Input.component/Input";
+
+
+
+class ContactPage extends React.Component {
+    constructor(props){
+        super(props);
+
+        this.state = {
```
Comment (chibuike07) [inline src/Pages/ContactPage/contact.jsx:21]: The state should be declared inside the nonregistrationcontext

File: src/Pages/ContactPage/contact.jsx [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+            message: "",
+            disabled: false,
+            messageSent : null,
+
+        };
+        this.handleChange = this.handleChange.bind(this);
+        this.handleSubmit = this.handleSubmit.bind(this);
+    }
+
+    handleChange = (e) => {
```
Comment (chibuike07) [inline src/Pages/ContactPage/contact.jsx:34]: Adjust the handle change function

File: src/Pages/ContactPage/contact.jsx [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+        this.setState({
+            [name] : value,
+        });
+    }
+    handleSubmit = (e) => {
+        e.preventDefault();
+        this.setState({
+            disabled : true,
+        })
+    };
```
Comment (chibuike07) [inline src/Pages/ContactPage/contact.jsx:48]: Handle submit function should be setting the state inside the context provider

**Review by chibuike07 — COMMENTED** [review]
The footer component is affected by your styles. Try as much as possible not to be passing props, we already have the context that gives us a unidirectional flow of the state.

**Review by emmyclef — COMMENTED** [review]
ok, just slight modifications needed.

**Review by emmyclef — APPROVED** [review]

**Review by HENRYKC24 — APPROVED** [review]

**Review by chibuike07 — APPROVED** [review]

**Comment by chibuike07** [comment]
This page has been approved

**Pull request closed without merging**